Work on UI testing guide, minor fixes (#1311)
feldera/feldera: docs/contributors/ui-testing.md

# Web Console testing

Regression UI testing is performed in the form of end-to-end integration tests using Playwright framework.

## e2e testing with Playwright
Existing Playwright tests are executed during CI
and can be run manually within provided devcontainer environment.

### Running
Run `yarn test`
to execute all tests on all supported platforms in background, or run `yarn test:ui` to open a UI to run tests interactively.
Add environment variable `CI=true` when executing tests in CI setting.

### Contributing tests

Regression testing is performed by comparing snapshots with the test results.
Snapshots are expected in the directory `feldera/web-console/playwright-snapshots` and are generated during test run if they are not found.
Snapshots are not a part of the feldera repo, and are instead stored at https://github.com/feldera/playwright-snapshots

> During the CI snapshots of the correct version are copied into the build directory.

Reviewer: ```suggestion
During CI, snapshots of the correct version are copied into the build directory.
```

> Environment variable `PLAYWRIGHT_SNAPSHOTS_COMMIT` in `feldera/.arg` specifies commit hash that is being used in CI to test against.

Reviewer: ```suggestion
Environment variable `PLAYWRIGHT_SNAPSHOTS_COMMIT` in `feldera/.arg` specifies the commit hash being used in CI to test against.
```

When committing new tests or updating screenshots for existing tests, `PLAYWRIGHT_SNAPSHOTS_COMMIT` needs to be updated as well.

> When testing locally, you need to manually clone `playwright-snapshots` at the correct commit hash

Reviewer: ```suggestion
When testing locally, you need to manually clone `playwright-snapshots` and checkout the correct commit hash.
```

Reviewer: Maybe add command to do that e.g., `git clone $REPO && git checkout $REV`

### Writing tests

If you want to use `Playwright codegen` to automatically create new tests from UI interactions,

> install Playwright on your host system.

Reviewer: URL for where to find install instructions
